test/gtfs/block_test.exs

```diff
@@ -19,42 +19,62 @@ defmodule Gtfs.BlockTest do
     ]
   }
 
-  test "can group trips and then get them" do
-    assert [@trip] ==
-             [@trip]
-             |> Block.group_trips_by_block()
-             |> Block.get(@trip.block_id, @trip.service_id)
-  end
+  describe "group_trips_by_block/ and get/3 " do
+    test "can group trips and then get them" do
+      assert [@trip] ==
+               [@trip]
+               |> Block.group_trips_by_block()
+               |> Block.get(@trip.block_id, @trip.service_id)
+    end
 
-  test "sorts trips by time" do
-    trips = [
-      %{
-        @trip
-        | id: "t2",
-          stop_times: [
-            %StopTime{stop_id: "s", time: 2}
-          ]
-      },
-      %{
-        @trip
-        | id: "t1",
-          stop_times: [
-            %StopTime{stop_id: "s", time: 1}
-          ]
-      }
-    ]
+    test "sorts trips by time" do
+      trips = [
+        %{
+          @trip
+          | id: "t2",
+            stop_times: [
+              %StopTime{stop_id: "s", time: 2}
+            ]
+        },
+        %{
+          @trip
+          | id: "t1",
+            stop_times: [
+              %StopTime{stop_id: "s", time: 1}
+            ]
+        }
+      ]
+
+      assert [%Trip{id: "t1"}, %Trip{id: "t2"}] =
+               trips
+               |> Block.group_trips_by_block()
+               |> Block.get("b", "service")
+    end
+
+    test "ignores trips without stop times" do
+      trips = [
+        %{@trip | stop_times: []}
+      ]
 
-    assert [%Trip{id: "t1"}, %Trip{id: "t2"}] =
-             trips
-             |> Block.group_trips_by_block()
-             |> Block.get("b", "service")
+      assert Block.group_trips_by_block(trips) == %{}
+    end
   end
 
-  test "ignores trips without stop times" do
-    trips = [
-      %{@trip | stop_times: []}
-    ]
+  describe "start_time/1" do
+    test "returns the time of the first stop of the first trip for this block" do
+      assert Block.start_time(block()) == 1
+    end
+  end
+
+  describe "end_time/1" do
+    test "returns the time of the last stop of the last trip for this block" do
+      assert Block.end_time(block()) == 2
```

Comment:
This can just be `[@trip]`, and `block()` isn't needed. The other tests go through the pipeline that you've abstracted into `block()` because they're testing `group_trips_by_block`, not because the pipeline necessary in order to get a block.

Reply:
👍 thanks